test_tests/test_selection_framework.sh

```diff
@@ -0,0 +1,20 @@
+#!/bin/bash
```

Comment:
I think you can create `main_test.go` and load these yamls directly from Go and do the testing.

This will allow to get code coverage right away, which is useful to see if you left any code path out of the tests.

Are you familiar with how go testing works? https://golang.org/pkg/testing/ 

This will require refactoring the `main()` into more specialized functions that you can call from the tests. The main should probably just parse command-line arguments and call some sort of `Run(args)` which then does:

```
readTestFile()
validate()
RunTests()
PrintResults()
```

kind of thing.